### PR TITLE
[Prefix] Added job to generate verified auxiliary file

### PR DESCRIPTION
**Pull request by loic-sharma**
Fixes https://github.com/NuGet/NuGetGallery/issues/4381

### CODE REVIEW

File: src/Search.GenerateAuxiliaryData/SqlScripts/VerifiedPackages.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+﻿SELECT	[PackageRegistrations].[Id] 'Id'
+FROM	[PackageRegistrations] (NOLOCK)
+WHERE   [PackageRegistrations].[IsVerified] = 1
```
Comment (shishirx34) [inline src/Search.GenerateAuxiliaryData/SqlScripts/VerifiedPackages.sql]: column name is "Verified"

**Comment by shishirx34** [comment]
please hold off on merging this until I merge the 'feature-prefix' branch to dev. This would need the schema migration applied to the databases to run successfully. I plan to merge that branch after merging my current PR.

File: src/Search.GenerateAuxiliaryData/VerifiedPackagesExporter.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            string defaultVerifiedPackagesScript,
+            string defaultName)
+            : base(logger, defaultConnectionString, defaultDestinationContainer, defaultName)
+        {
+            _verifiedPackagesScript = defaultVerifiedPackagesScript;
+        }
+
+        protected override JContainer GetResultOfQuery(SqlConnection connection)
+        {
+            var rankingsTotalCommand = new SqlCommand(GetEmbeddedSqlScript(_verifiedPackagesScript), connection);
```
Comment (shishirx34) [inline src/Search.GenerateAuxiliaryData/VerifiedPackagesExporter.cs]: rename to `verifiedPackagesCommand` or something appropriate?

File: src/Search.GenerateAuxiliaryData/VerifiedPackagesExporter.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+using System;
+using System.Data;
+using System.Data.SqlClient;
+using Microsoft.Extensions.Logging;
+using Microsoft.WindowsAzure.Storage.Blob;
+using Newtonsoft.Json.Linq;
+
+namespace Search.GenerateAuxiliaryData
+{
+    // Public only to facilitate testing.
```
Comment (chenriksson) [inline src/Search.GenerateAuxiliaryData/VerifiedPackagesExporter.cs]: Regarding comment - are there tests for this code? If so, can we not use internalsVisibleTo?

File: src/Search.GenerateAuxiliaryData/VerifiedPackagesExporter.cs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+            var colNames = GetColMappingFromSqlDataReader(reader);
+            var result = new JArray();
+
+            while (reader.Read())
+            {
+                result.Add(reader.GetString(colNames[_colPackageId]));
+            }
+
+            reader.Close();
```
Comment (chenriksson) [inline src/Search.GenerateAuxiliaryData/VerifiedPackagesExporter.cs]: Would put Close call in a finally block

File: src/Search.GenerateAuxiliaryData/VerifiedPackagesExporter.cs (hunk 3 above)
Comment (loic-sharma) [inline src/Search.GenerateAuxiliaryData/VerifiedPackagesExporter.cs]: This was a copy-paste mistake. I will remove this comment, make the call internal, and add a test.

**Comment by shishirx34** [comment]
My changes are merged into DEV, feel free to merge these changes in. Please update the deployment task to make sure it clearly reflects the order of deployments needed to deploy these changes - https://github.com/NuGet/NuGetGallery/issues/4591

**Pull request merged**